### PR TITLE
feat(IoTDA): add a resource to manage AMQP queue

### PR DESCRIPTION
**Pull request by hmbig2**
**What this PR does / why we need it**:
add a resource to manage AMQP queue

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [X] Tests added/passed.
* [X] Documentation updated.
* [X] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/iotda' TESTARGS='-run=TestAccAmqp_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/iotda -v -run=TestAccAmqp_basic -timeout 360m -parallel 4
=== RUN   TestAccAmqp_basic
--- PASS: TestAccAmqp_basic (13.18s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/iotda     13.283s
```

### CODE REVIEW

File: huaweicloud/services/acceptance/iotda/resource_huaweicloud_iotda_amqp_test.go [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+	name := acceptance.RandomAccResourceName()
+	rName := "huaweicloud_iotda_amqp.test"
+
+	rc := acceptance.InitResourceCheck(
+		rName,
+		&obj,
+		getAmqpResourceFunc,
+	)
+
+	resource.Test(t, resource.TestCase{
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/acceptance/iotda/resource_huaweicloud_iotda_amqp_test.go]: please use ParallelTest

File: docs/resources/iotda_amqp.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+resource "huaweicloud_iotda_amqp" "queue" {
+  name = "queue_name"
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `region` - (Optional, String, ForceNew) Specifies the region in which to create the IoTDA resource space resource.
```
Comment (Jason-Zhang9309) [inline docs/resources/iotda_amqp.md]: s/IoTDA resource space/IoTDA AMQP queue

File: huaweicloud/services/iotda/resource_huaweicloud_iotda_amqp.go [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+	if err != nil {
+		return diag.Errorf("error creating IoTDA v5 client: %s", err)
+	}
+
+	createOpts := model.AddQueueRequest{
+		Body: &model.QueueInfo{
+			QueueName: d.Get("name").(string),
+		},
+	}
+	log.Printf("[DEBUG] Create IoTDA AMQP params: %#v", createOpts)
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/iotda/resource_huaweicloud_iotda_amqp.go]: please use AMQP queue to keep same with the doc

File: huaweicloud/services/iotda/resource_huaweicloud_iotda_amqp.go [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+	if err != nil {
+		return diag.Errorf("error creating IoTDA v5 client: %s", err)
+	}
+
+	detail, err := client.ShowQueue(&model.ShowQueueRequest{QueueId: d.Id()})
+	if err != nil {
+		return common.CheckDeletedDiag(d, err, "error retrieving IoTDA AMQP")
+	}
+
+	if detail != nil && detail.QueueId == nil {
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/iotda/resource_huaweicloud_iotda_amqp.go]: please use `detail == nil || detail.QueueId == nil`

File: huaweicloud/services/acceptance/iotda/resource_huaweicloud_iotda_amqp_test.go [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+)
+
+func getAmqpResourceFunc(conf *config.Config, state *terraform.ResourceState) (interface{}, error) {
+	client, err := conf.HcIoTdaV5Client(acceptance.HW_REGION_NAME)
+	if err != nil {
+		return nil, fmt.Errorf("error creating IoTDA v5 client: %s", err)
+	}
+
+	detail, err := client.ShowQueue(&model.ShowQueueRequest{QueueId: state.Primary.ID})
+	if detail != nil && detail.QueueId == nil {
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/acceptance/iotda/resource_huaweicloud_iotda_amqp_test.go]: please use `detail == nil || detail.QueueId == nil`

File: docs/resources/iotda_amqp.md (hunk 2 above)
Comment (hmbig2) [reply docs/resources/iotda_amqp.md]: fix

File: huaweicloud/services/acceptance/iotda/resource_huaweicloud_iotda_amqp_test.go (hunk 5 above)
Comment (hmbig2) [reply huaweicloud/services/acceptance/iotda/resource_huaweicloud_iotda_amqp_test.go]: fix

File: huaweicloud/services/iotda/resource_huaweicloud_iotda_amqp.go (hunk 4 above)
Comment (hmbig2) [reply huaweicloud/services/iotda/resource_huaweicloud_iotda_amqp.go]: fix

File: huaweicloud/services/iotda/resource_huaweicloud_iotda_amqp.go (hunk 3 above)
Comment (hmbig2) [reply huaweicloud/services/iotda/resource_huaweicloud_iotda_amqp.go]: ok

File: huaweicloud/services/acceptance/iotda/resource_huaweicloud_iotda_amqp_test.go (hunk 1 above)
Comment (hmbig2) [reply huaweicloud/services/acceptance/iotda/resource_huaweicloud_iotda_amqp_test.go]: done

**Pull request merged**